roadmap/implementers-guide/src/types/overseer-protocol.md

```diff
@@ -236,6 +236,30 @@ enum ChainApiMessage {
 		/// The response channel.
 		response_channel: ResponseChannel<Result<Vec<Hash>, Error>>,
 	}
+	/// Blacklist a set of block hashes to be considered poisened.
+	/// Assumes all hashes exist only once in the set.
+	Blacklist{
+		set: Vec<Hash>,
+		response: ResponseChannel<Result<()>>,
+	},
+	/// Collects all hashes, that are part from a given hash up to
+	/// the current active heads.
+	Descendants {
```

Comment:
How would this be implemented? I'm not sure if client's db layout allows for such queries.

Reply:
Worst case, a block number has to be specified and backtracking from all leaves is to be done in lockstep, since each block must have knowledge of its parent.

Reply:
Actually I'm not sure we need this call at all. Once we blacklist a block, I'd expect that the client will automatically reject all forks that are based on it as a general fork-choice rule, but need to double check that.